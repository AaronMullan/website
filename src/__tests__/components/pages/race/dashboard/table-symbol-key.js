import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import TableSymbolKey from '~components/pages/race/dashboard/table-symbol-key'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    disparityNote: {
      contentful_id: 'aNeWRaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More disparity snippet content</p>',
        },
      },
    },
    comparibleNote: {
      contentful_id: 'RaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More comparible snippet content</p>',
        },
      },
    },
  }))
})

describe('Components : Pages : Race : Dashboard : Table Symbol Key', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TableSymbolKey state="NV" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
