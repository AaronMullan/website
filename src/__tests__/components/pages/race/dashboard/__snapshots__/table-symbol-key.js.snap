// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Race : Dashboard : Table Symbol Key renders correctly 1`] = `
<div
  className="container"
>
  <p
    className="key"
  >
    <span
      className="disparitySymbol inKey"
    >
      <span
        className="a11y-only"
      >
        Figure is significantly higher than population proportion
      </span>
      <span
        className="inner"
      />
    </span>
    Racial/ethnic disparity likely.
     
    <a
      className="js-disabled"
      href="#notes-disparity"
    >
      See why 
      <span
        aria-hidden={true}
      >
        ↓
      </span>
    </a>
    <button
      aria-controls="table-symbol-disparity-nv"
      aria-expanded={false}
      className="disclosureButton js-enabled"
      onClick={[Function]}
    >
      <span
        className="text"
      >
        See why
      </span>
       
      <span
        aria-hidden={true}
      >
        ↓
      </span>
    </button>
  </p>
  <p
    className="key"
  >
    <span
      className="caution inKey"
    >
      <span
        className="a11y-only"
      >
        This data is not reliable
      </span>
    </span>
    Should not be compared with percentage of population.
     
    <a
      className="js-disabled"
      href="#notes-comparible"
    >
      See why 
      <span
        aria-hidden={true}
      >
        ↓
      </span>
    </a>
    <button
      aria-controls="table-symbol-notComparible-nv"
      aria-expanded={false}
      className="disclosureButton js-enabled"
      onClick={[Function]}
    >
      <span
        className="text"
      >
        See why
      </span>
       
      <span
        aria-hidden={true}
      >
        ↓
      </span>
    </button>
  </p>
  <div />
  <div
    className="disclosurePane"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h2>Content</h2><p>More disparity snippet content</p>",
      }
    }
    data-expanded={false}
    hidden={true}
    id="table-symbol-disparity-nv"
  />
  <div
    className="disclosurePane"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h2>Content</h2><p>More comparible snippet content</p>",
      }
    }
    data-expanded={false}
    hidden={true}
    id="table-symbol-notComparible-nv"
  />
</div>
`;
